integration-test-remote/src/test/java/io/cdap/cdap/app/etl/PubSubTest.java

```diff
@@ -0,0 +1,341 @@
+  private static SubscriptionAdminClient subscriptionAdmin;
+  private ArrayList<ProjectTopicName> topicsToDelete;
+  private ArrayList<ProjectSubscriptionName> subscriptionsToDelete;
+  private String googleCloudVersion;
+
+  @BeforeClass
+  public static void testClassSetup() throws IOException {
+    topicAdmin = TopicAdminClient.create(
+      TopicAdminSettings.newBuilder().setCredentialsProvider(getCredentialProvider()).build());
+    subscriptionAdmin = SubscriptionAdminClient.create(
+      SubscriptionAdminSettings.newBuilder().setCredentialsProvider(getCredentialProvider()).build());
+  }
+
+  private static CredentialsProvider getCredentialProvider() {
+    return () -> GoogleCredentials.fromStream(
+      new ByteArrayInputStream(getServiceAccountCredentials().getBytes(StandardCharsets.UTF_8)));
+  }
+
+  void checkPluginExists(String pluginName, String pluginType, String artifact) {
+    Preconditions.checkNotNull(pluginName);
+    Preconditions.checkNotNull(pluginType);
+    Preconditions.checkNotNull(artifact);
+
+    try {
+      Tasks.waitFor(true, () -> {
+        try {
+          final ArtifactId artifactId = TEST_NAMESPACE.artifact(artifact, version);
+          List<PluginSummary> plugins =
+            artifactClient.getPluginSummaries(artifactId, pluginType, ArtifactScope.SYSTEM);
+          return plugins.stream().anyMatch(pluginSummary -> pluginName.equals(pluginSummary.getName()));
+        } catch (ArtifactNotFoundException e) {
+          // happens if the relevant artifact(s) were not added yet
+          return false;
+        }
+      }, 5, TimeUnit.MINUTES, 3, TimeUnit.SECONDS);
+    } catch (Exception e) {
+      Throwables.throwIfUnchecked(e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void innerSetup() throws Exception {
+    ImmutableList.of(ImmutableList.of(GOOGLE_SUBSCRIBER_PLUGIN_NAME, StreamingSource.PLUGIN_TYPE, "cdap-data-streams"),
+                     ImmutableList.of(GOOGLE_PUBLISHER_PLUGIN_NAME, BatchSink.PLUGIN_TYPE, "cdap-data-pipeline"))
+      .forEach((pluginInfo) -> checkPluginExists(pluginInfo.get(0), pluginInfo.get(1), pluginInfo.get(2)));
+
+    ArtifactSummary googleCloudArtifact = artifactClient
+      .listVersions(TEST_NAMESPACE, "google-cloud", ArtifactScope.SYSTEM)
+      .stream().findFirst().orElse(null);
+
+    if (googleCloudArtifact != null) {
+      googleCloudVersion = googleCloudArtifact.getVersion();
+    } else {
+      throw new ArtifactNotFoundException(TEST_NAMESPACE.getNamespace(), "google-cloud");
+    }
+
+    topicsToDelete = new ArrayList<>();
+    subscriptionsToDelete = new ArrayList<>();
+  }
+
+  @Override
+  public void innerTearDown() {
+    for (ProjectTopicName topic : topicsToDelete) {
+      try {
+        topicAdmin.deleteTopic(topic);
+        LOG.info("Deleted topic {}", topic.toString());
+      } catch (ApiException ex) {
+        LOG.warn("Failed to delete topic {}", topic.toString());
```

Comment:
I think you should also log error stacktrace here

Reply:
Thanks, changed.